feat: implement cross-platform build support for jib builder in skaffold

gsquared94 opened the pull request:
<!-- Thank you for your contribution! -->

<!-- Include if applicable: -->
Related: #6120 <!-- tracking issues that this PR will close -->

tejal29 commented:
Can you please add some instructions to verify in the PR description?

tejal29 commented on pkg/skaffold/build/jib/gradle.go:
```diff
@@ -97,15 +99,22 @@ func getSyncMapCommandGradle(ctx context.Context, workspace string, a *latestV1.
-	args := gradleBuildArgsFunc(task, a, skipTests, showColors, MinimumJibGradleVersion)
+func GenerateGradleBuildArgs(task, imageName string, a *latestV1.JibArtifact, platforms platform.Matcher, skipTests, pushImages bool, deps []*latestV1.ArtifactDependency, r ArtifactResolver, insecureRegistries map[string]bool, showColors bool) []string {
+	minVersion := MinimumJibGradleVersion
+	if platforms.IsMultiPlatform() {
+		minVersion = MinimumJibGradleVersionForMultiPlatform
+	}
```
Is cross platform builds ie. when `len(platforms)==1` is support by `MinimumJibGradleVersion` ?

gsquared94 replied:
fixed

tejal29 reviewed (commented):
One suggestion.
Looks like the minimum version check is incorrect for cross platform build case. 
Please add steps to verify!

gsquared94 commented:
> One suggestion. Looks like the minimum version check is incorrect for cross platform build case. Please add steps to verify!

thanks for the review @tejal29 . I've updated the PR description with testing instructions and fixed the above issue. PTAL.
